internal/rbd/encryption.go

```diff
@@ -437,3 +437,72 @@ func (ri *rbdImage) RemoveDEK(ctx context.Context, volumeID string) error {
 
 	return nil
 }
+
+// GetEncryptionPassphraseSize returns the value of `encryptionPassphraseSize`.
+func GetEncryptionPassphraseSize() int {
+	return encryptionPassphraseSize
+}
+
+// RotateKey processes the key rotation for the RBD Volume.
+func (rv *rbdVolume) RotateEncryptionKey(ctx context.Context) error {
+	if !rv.isBlockEncrypted() {
+		return errors.New("key rotation not supported for the encryption type")
+	}
+
+	// Verify that the underlying device has been setup for encryption
+	currState, err := rv.checkRbdImageEncrypted(ctx)
```

Comment:
Adding the check here would require us to change the spec to add support for volume capabilities.

Reply:
we cannot add a check on the csi-addons, we can document this as an unsupported feature (this can lead to other problem as we are supporting auto rotation) or change the spec to add this one. 

@nixpanic WDYT?

Reply:
If we decide to add it, a PR is open at: https://github.com/csi-addons/spec/pull/70